### PR TITLE
feat(menu): added danger menu item variation

### PR DESCRIPTION
**Pull request by blfetche**
Links to #5057 

image

image

### CODE REVIEW

File: src/patternfly/components/Menu/examples/Menu.md [hunk 1]
```diff
@@ -53,6 +53,44 @@ import './Menu.css'
...
+            {{#> menu-item-text}}
+              Action 1
+            {{/menu-item-text}}
+          {{/menu-item-main}}
+        {{/menu-item}}
+      {{/menu-list-item}}
+      {{#> menu-list-item}}
+        {{#> menu-item}}
+          {{#> menu-item-main}}
+            {{#> screen-reader}}Danger Item{{/screen-reader}}
```
Comment (mcoker) [inline src/patternfly/components/Menu/examples/Menu.md:73]: case and a colon - not sure if it matters, but that's what we use in the alert, modal, and popover.

```suggestion
            {{#> screen-reader}}Danger item:{{/screen-reader}}
```

File: src/patternfly/components/Menu/menu.scss [hunk 2]
```diff
@@ -627,6 +628,10 @@
   flex-grow: 1;
 }
 
+.pf-c-menu__item-text.pf-m-danger {
```
Comment (mcoker) [inline src/patternfly/components/Menu/menu.scss:631]: If you move this selector to `__list-item`, you can set this color by overriding the text color var in `__list-item&.pf-m-danger`. You'll just want to make sure it plays nicely with the disabled variation (text should be light grey if the item is danger and disabled)

`--pf-c-menu__list-item--Color: --pf-c-menu__list-item--m-danger__item--Color;`

**Review by mcoker — CHANGES_REQUESTED** [review]
Nice! Left some comments for review. Also since it came up in retro, here's a link to the commit message format we should follow: https://semantic-release.gitbook.io/semantic-release/

In short, use either:
* `fix()` to create a patch release (v1.2.3 -> 1.2.4) 
  * this is a general maintenance or fix change to an existing thing that will give users some new CSS that they can get in a new version of PF
* `feat()` to create a minor version release (v1.2.3 -> 1.3.0)
  * this is a new thing or an enhancement/feature to an existing thing that will give users some new CSS with a new version of PF
* `chore()` or `docs()` to commit something that doesn't create a new version of PF
  * Use this for docs updates, general updates to the examples and stuff fall under this, rework of handlebars code... basically anything that doesn't touch the SCSS or compiled CSS. Generally there is no reason to version the PF package from any of those changes.
  * These commits will be bundled with the next release - the next PR that has either `fix()` or `feat()` in the title/commit message.

Basically ask yourself if people who pull in patternfly via [`@patternfly/patternfly`\](https://www.npmjs.com/package/@patternfly/patternfly) will benefit from this change by getting the change in a new version of PF, and choose the commit prefix accordingly.

And the description should be in lower case, written in past tense. The commit title you use here will show up as a bullet point in our release notes - https://www.patternfly.org/v4/developer-resources/release-notes/html

So for this one, I'd just rewrite it to `feat(menu): added danger menu item variation` or something like that. And it's easiest just to edit the title here in GH using the "edit" button to the right of the title. 

Screen Shot 2022-09-19 at 4 17 42 PM

**Comment by blfetche** [comment]
Thank you for this clarification.

File: src/patternfly/components/Menu/examples/Menu.md [hunk 3]
```diff
@@ -53,6 +53,44 @@ import './Menu.css'
...
+      {{#> menu-list-item}}
+        {{#> menu-item}}
+          {{#> menu-item-main}}
+            {{#> menu-item-text}}
+              Action 1
+            {{/menu-item-text}}
+          {{/menu-item-main}}
+        {{/menu-item}}
+      {{/menu-list-item}}
+      {{#> menu-list-item}}
```
[the remark this answers is not in the document]
Comment (blfetche) [reply src/patternfly/components/Menu/examples/Menu.md:70]: Applied update related to this edit, thanks

File: src/patternfly/components/Menu/examples/Menu.md [hunk 4]
```diff
@@ -53,6 +53,44 @@ import './Menu.css'
...
+            {{/menu-item-text}}
+          {{/menu-item-main}}
+        {{/menu-item}}
+      {{/menu-list-item}}
+      {{> divider divider--type="li"}}
+      {{#> menu-list-item}}
+        {{#> menu-item}}
+          {{#> menu-item-main}}
+            {{#> menu-item-text}}
+              Action 3
```
[the remark this answers is not in the document]
Comment (blfetche) [reply src/patternfly/components/Menu/examples/Menu.md]: Applied update related to this edit, thanks

File: src/patternfly/components/Menu/examples/Menu.md (hunk 1 above)
Comment (blfetche) [reply src/patternfly/components/Menu/examples/Menu.md]: Applied update related to this edit, thanks

File: src/patternfly/components/Menu/menu.scss (hunk 2 above)
Comment (blfetche) [reply src/patternfly/components/Menu/menu.scss]: Applied update related to this edit, thanks

File: src/patternfly/components/Menu/menu.scss [hunk 5]
```diff
@@ -567,6 +568,10 @@
     padding-top: var(--pf-c-menu__list-item--m-loading--PaddingTop);
     overflow: hidden; // prevents spinner rotation from overflowing
   }
+
+  &.pf-m-danger {
+    --pf-c-menu__list-item--Color: var(--pf-c-menu__list-item--m-danger__item--Color, inherit);
```
Comment (mcoker) [inline src/patternfly/components/Menu/menu.scss:573]:
```suggestion
    --pf-c-menu__item--Color: var(--pf-c-menu__list-item--m-danger__item--Color, inherit);
```

small change - you want to make the item color red - similar to load. https://github.com/patternfly/patternfly/blob/736e343ceecb31893c00afeef3593c486e273437/src/patternfly/components/Menu/menu.scss#L557-L560

File: src/patternfly/components/Menu/menu.scss [hunk 6]
```diff
@@ -627,6 +632,7 @@
   flex-grow: 1;
 }
 
+
```
Comment (mcoker) [inline src/patternfly/components/Menu/menu.scss:635]: nit - looks like an extra blank line snuck in

```suggestion
```

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks great! One small tweak needed to the color we're changing. Also can you move this example to the end? Typically we add new things at the end unless they're associated with an existing feature. Probably makes sense to come after the "view more" and "loading" examples since they're basically item variations like this?

File: src/patternfly/components/Menu/menu.scss (hunk 5 above)
Comment (blfetche) [reply src/patternfly/components/Menu/menu.scss]: Corrected

File: src/patternfly/components/Menu/examples/Menu.md [hunk 7]
```diff
@@ -1782,7 +1820,8 @@ import './Menu.css'
 | `.pf-m-drilldown` | `.pf-c-menu` | Modifies the menu so that all nested `.pf-c-menu` elements "drill down". |
 | `.pf-m-current-path` | `.pf-c-menu.pf-m-drilldown .pf-c-menu__list-item` | Modifies the menu list item for current path state. |
 | `.pf-m-drilled-in` | `.pf-c-menu.pf-m-drilldown, .pf-c-menu.pf-m-drilldown .pf-c-menu` | Modifies the menu list for drilled in state. |
-| `.pf-m-static` | `.pf-c-menu.pf-m-drilldown .pf-c-menu` | Modifies the menu list for drilled static state. |
+| `.pf-m-static` | `.pf-c-menu.pf-m-drilldown .pf-c-menu` | Modifies the menu list for drilled static state. | 
+| `.pf-m-danger` | `pf-c-menu__item-text` | Modifies the menu item text to indicate danger styling. |
```
Comment (mcoker) [inline src/patternfly/components/Menu/examples/Menu.md:1824]: updated this to match the load/loading modifiers.

```suggestion
| `.pf-m-danger` | `.pf-c-menu__list-item` | Modifies a list item for danger styles. |
```

**Review by mcoker — CHANGES_REQUESTED** [review]
Just one small docs nit and LGTM!

File: src/patternfly/components/Menu/examples/Menu.md (hunk 7 above)
Comment (blfetche) [reply src/patternfly/components/Menu/examples/Menu.md:1824]: Thanks again for these reviews. Definitely helping me along.

**Review by mcoker — APPROVED** [review]
Woo!
